docs/en/latest/plugins/consumer-restriction.md

```diff
@@ -116,6 +117,80 @@ HTTP/1.1 403 Forbidden
 {"message":"The consumer_name is forbidden."}
 ```
 
+### How to restrict `allowed_methods`
+
+This example restrict the user `jack1` to only `POST` on the resource
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "uri": "/index.html",
+    "upstream": {
+        "type": "roundrobin",
+        "nodes": {
+            "127.0.0.1:1980": 1
+        }
+    },
+    "plugins": {
+        "basic-auth": {},
+        "consumer-restriction": {
+            "whitelist": [
```

Comment:
```
        "consumer-restriction": {
            "allowed_by_methods":[{
                "user": "jack1",
                "methods": ["POST"]
            }]
        }
```
is enough. There is no need to configure `jack1` in two places. And the `allow` already has the meaning of `whitelist`.

Reply:
Ok so i completely drop the whitelist part in favor of allowed_by_methods ?

Reply:
If so that would mean some rework of the whole plugin.
Right now the `allowed_by_methods` required to set the list of methods , so if we drop `whitelist`we need a quick setup . Like if there is no method , then full access ? or the user still ha to set the list of method but we take into account a `*` wildcard ?

Reply:
OK. I made a mistake, the `whitelist` is not required with `allowed_by_methods`. I think we should show them separately in the doc and test.

Reply:
I think we can use the schema below to allow both `whitelist` and `allowed_by_methods`:
```lua
local schema = {
    type = "object",
    properties = {
        type = {
            type = "string",
            enum = {"consumer_name", "service_id"},
            default = "consumer_name"
        },
        blacklist = {
            type = "array",
            minItems = 1,
            items = {type = "string"}
        },
        whitelist = {
            type = "array",
            minItems = 1,
            items = {type = "string"}
        },
        allowed_by_methods = {
            type = "array",
            items = {
                type = "object",
                properties = {
                    user = {
                        type = "string"
                    },
                    methods = {
                        type = "array",
                        minItems = 1,
                        items = {
                            type = "string",
                            enum = {"GET", "POST", "PUT", "DELETE", "PATCH", "HEAD",
                                    "OPTIONS", "CONNECT", "TRACE"},
                        }
                    }
                }
            }
        },
        rejected_code = {type = "integer", minimum = 200, default = 403}
    },
    anyOf = {
        {required = {"blacklist"}},
        {required = {"whitelist"}},
        {required = {"allowed_by_methods"}}
    },
}

```

Reply:
We can change the title of https://github.com/apache/apisix/blob/29b1e2dfa4ae27cb8fff0511d30106803224d30c/t/plugin/consumer-restriction.t#L57 and say that the priority is blacklist > whitelist > allowed_by_methods.

Reply:
It is a large change but I think it is flexible for the users.

Reply:
Ok , no problem.
So `allowed_by_methods` could exist without `whitelist`.
If whitelist is set it will override what could be specified in the `allowed_by_methods` section